lib/services/elasticsearch.js

```diff
@@ -1201,15 +1222,18 @@ class ElasticSearch extends Service {
   _mexecute(esRequest, documents, partialErrors) {
     assertWellFormedRefresh(esRequest);
 
-    if (!this.kuzzle.indexCache.exists(esRequest.index, esRequest.type)) {
-      return Bluebird.reject(new PreconditionError(`Index '${esRequest.index}' and/or collection '${esRequest.type}' don't exist`));
-    }
+    return this.kuzzle.indexCache.exists(esRequest.index, esRequest.type)
+      .then(exists => {
+        if (! exists) {
+          throw new PreconditionError(`Index '${esRequest.index}' and/or collection '${esRequest.type}' does not exist`);
+        }
 
-    if (documents.length > this.kuzzle.config.limits.documentsWriteCount) {
-      return Bluebird.reject(new SizeLimitError(`Number of documents exceeds the server configured value (${this.kuzzle.config.limits.documentsWriteCount})`));
-    }
+        if (documents.length > this.kuzzle.config.limits.documentsWriteCount) {
+          throw new SizeLimitError(`Number of documents exceeds the server configured value (${this.kuzzle.config.limits.documentsWriteCount})`);
+        }
 
-    return (documents.length > 0 ? this.client.bulk(esRequest) : Bluebird.resolve({items: []}))
+        return (documents.length > 0 ? this.client.bulk(esRequest) : Bluebird.resolve({items: []}))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove those useless parentheses. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AParentheses)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a semicolon at the end of this statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ASemicolon)